pallets/settlement/src/lib.rs

```diff
@@ -1168,6 +1359,120 @@ impl<T: Trait> Module<T> {
         )
     }
 
+    fn execute_confidential_instruction(instruction_id: u64) -> (u32, Weight) {
```

Comment:
1. Needs a doc comment.

2. Having a struct as the return type would save a bit of refactoring if the content of the type changes. Also having field names would give extra info as opposed to `.0` and `.1`.

Reply:
This was a duplicate of the `execute_instruction` function. I have removed it. Your comment is still valid and could be good refactoring task for core team :)